common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDB.java

```diff
@@ -229,6 +241,10 @@ public long count(Class<?> type, String index, Object indexedValue) throws Excep
     return idx.getCount(idx.end(null, indexedValue));
   }
 
+  /**
+   * Trying to close a JNI LevelDB handle with a closed DB can cause JVM crashes,
```

Comment:
nit: Trying to close a JNI LevelDB handle with a closed DB can cause JVM crashes`. T`his ensures that all iterators are correctly closed before DB is closed.

Btw, I guess this comment looks better than the one for `iteratorTracker`. We may just need to have one in `iteratorTracker` with replacing the sentences, and maybe better to add explanation of why we use soft reference to track these iterators.

Reply:
Thanks, updated.